File: ObjectiveC/FlexChart101/FlexChart101/ScrollingController.m
//
//  ScrollingController.m
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "ScrollingController.h"
@import XuniFlexChartDynamicKit;
#import "WeatherData.h"

@interface ScrollingController ()
@property (weak, nonatomic) IBOutlet FlexChart *chart;

@end

@implementation ScrollingController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    FlexChart *chart = self.chart;
    NSMutableArray *chartData = [WeatherData demoData];
    
    XuniSeries *precip = [[XuniSeries alloc] initForChart:chart binding:@"precipitation" name:@"Precip"];
    XuniSeries *temp= [[XuniSeries alloc] initForChart:chart binding:@"temp" name:@"Temp"];
    temp.chartType = XuniChartTypeSpline;
    XuniSeries *volume= [[XuniSeries alloc] initForChart:chart binding:@"volume" name:@"Volume"];
    volume.chartType = XuniChartTypeSpline;
    
    XuniAxis *axisT = [[XuniAxis alloc] initWithPosition:XuniPositionRight forChart:chart];
    axisT.min = @20;
    axisT.max = @80;
    axisT.majorUnit = 5;
    axisT.title = @"Volume";
    axisT.axisLineVisible = NO;
    axisT.majorGridVisible = NO;
    axisT.majorTickWidth = 0;
    axisT.labelsVisible = YES;
    [chart.axesArray addObject:axisT];

    chart.chartType = XuniChartTypeColumn;
    chart.bindingX = @"index";
    chart.itemsSource = chartData;
    chart.zoomMode = XuniZoomModeX;
    chart.axisY.min = @4;
    chart.axisY.max = @20;
    chart.axisY.majorUnit = 2;
    chart.axisY.axisLineVisible = NO;
    chart.axisY.majorTickWidth = 0;
    chart.axisX.labelAngle = 90;
    chart.axisX.majorGridVisible = NO;
    chart.axisX.displayedRange = 10;
    chart.palette = [XuniPalettes midnight];
    chart.header = @"Drag to scroll/Pinch to zoom";
    chart.headerFont = [UIFont systemFontOfSize:14];
    chart.headerTextAlignment = XuniHorizontalAlignmentCenter;
    
    [temp setAxisY:axisT];
    [volume setAxisY:axisT];
    [chart.series addObject:precip];
    [chart.series addObject:temp];
    [chart.series addObject:volume];
    
}



@end

File: Swift/FlexChart101/FlexChart101/ScrollingController.swift
//
//  ScrollingController.swift
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

import UIKit
import XuniFlexChartKit

class ScrollingController: UIViewController {
    
    var _chart = FlexChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = NSLocalizedString("Scrolling", comment: "")
        
        // Do any additional setup after loading the view.
        let precip = XuniSeries(forChart: _chart, binding: "precipitation", name: "Precip")
        let temp = XuniSeries(forChart: _chart, binding: "temp", name: "Temp")
        temp.chartType = XuniChartType.Spline
        let volume = XuniSeries(forChart: _chart, binding: "volume", name: "Volume")
        volume.chartType = XuniChartType.Spline
        
        let axisT = XuniAxis(position: XuniPosition.Right, forChart: _chart)
        axisT.min = 20
        axisT.max = 80
        axisT.majorUnit = 5
        axisT.title = "Volume"
        axisT.axisLineVisible = false
        axisT.majorGridVisible = false
        axisT.majorTickWidth = 0
        axisT.labelsVisible = true
        _chart.axesArray.addObject(axisT)
        
        _chart.chartType = XuniChartType.Column
        _chart.bindingX = "index"
        _chart.itemsSource = WeatherData.demoData()
        _chart.zoomMode = XuniZoomMode.X
        _chart.axisY.min = 4
        _chart.axisY.max = 20
        _chart.axisY.majorUnit = 2
        _chart.axisY.axisLineVisible = false
        _chart.axisY.majorTickWidth = 0
        _chart.axisX.labelAngle = 90
        _chart.axisX.majorGridVisible = false
        _chart.axisX.displayedRange = 10
        _chart.legend.position = XuniChartLegendPosition.Auto
        _chart.legend.orientation = XuniChartLegendOrientation.Auto
        _chart.palette = XuniPalettes.midnight()
        _chart.header = NSLocalizedString("Drag to scroll/Pinch to zoom", comment: "")
        _chart.headerFont = UIFont.systemFontOfSize(14)
        _chart.headerTextAlignment = XuniHorizontalAlignment.Center
        
        temp.axisY = axisT
        volume.axisY = axisT
        _chart.series.addObject(precip)
        _chart.series.addObject(temp)
        _chart.series.addObject(volume)

        self.view.addSubview(_chart)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        _chart.frame = CGRectMake(0, 65, self.view.bounds.size.width, self.view.bounds.size.height - 65)
    }
    
    /*
    // MARK: - Navigation
    
    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
    // Get the new view controller using segue.destinationViewController.
    // Pass the selected object to the new view controller.
    }
    */
    
}

